desc/profiles.py

```diff
@@ -876,15 +876,104 @@ def compute(self, grid, params=None, dr=0, dt=0, dz=0):
             values of the profile or its derivative at the points specified
 
         """
-        if params is None:
-            params = self.params
         if dt != 0 or dz != 0:
             return jnp.zeros_like(grid.nodes[:, 0])
-        x = self.knots
-        f = params
-        xq = grid.nodes[:, 0]
-        fq = interp1d(xq, x, f, method=self._method, derivative=dr, extrap=True)
-        return fq
+        params = setdefault(params, self._params)
+        return interp1d(
+            xq=grid.nodes[:, 0],
+            x=self._knots,
+            f=params,
+            method=self._method,
+            derivative=dr,
+            extrap=True,
+        )
+
+
+class HermiteSplineProfile(_Profile):
+    """Profile represented by a piecewise cubic Hermite spline.
+
+    Parameters
+    ----------
+    r : array-like
+        1-D array containing values of the independent variable.
+        Values must be real, finite, and in strictly increasing order in [0, 1].
+    f: array-like
+        Array containing values of the dependent variable.
+    dfdr: array-like
+        Array containing derivatives of the dependent variable.
+    name : str
+        Optional name of the profile.
+
+    """
+
+    _io_attrs_ = _Profile._io_attrs_ + ["_knots", "_params"]
+
+    def __init__(self, r, f, dfdr, name=""):
+        super().__init__(name)
+        r, f, dfdr = jnp.atleast_1d(r, f, dfdr)
+        self._knots = r
+        self._params = jnp.stack([f, dfdr])
```

Comment:
Everywhere else we assume `params` is a 1d array so I think this should be `concatenate` not `stack`